tests/integration/container/tests/custom_endpoint.test.ts

```diff
@@ -0,0 +1,345 @@
+  const start = Date.now();
+
+  const timeoutEndMs = Date.now() + 1200000; // 20 minutes
+  let hasCorrectState = false;
+  while (Date.now() < timeoutEndMs) {
+    const input = {
+      DBClusterEndpointIdentifier: endpointId
+    };
+    const command = new DescribeDBClusterEndpointsCommand(input);
+    const result = await rdsClient.send(command);
+    const endpoints = result.DBClusterEndpoints;
+    if (endpoints.length !== 1) {
+      fail(
+        `Unexpected number of endpoints returned while waiting for custom endpoint to have the specified list of members. Expected 1, got ${endpoints.length}.`
+      );
+    }
+
+    const endpoint = endpoints[0];
+    membersList.sort();
+    endpoint.StaticMembers.sort();
+    hasCorrectState = endpoint.Status === "available" && arraysAreEqual(membersList, endpoint.StaticMembers);
+    if (hasCorrectState) {
+      break;
+    }
+
+    await sleep(3000);
+  }
+
+  if (!hasCorrectState) {
+    fail(`Timed out while waiting for the custom endpoint to stabilize: '${endpointId}'.`);
+  }
+
+  logger.info(`waitUntilEndpointHasMembers took ${(Date.now() - start) / 1000} seconds`);
+}
+
+function arraysAreEqual(array1: any[], array2: any[]): boolean {
+  if (array1.length !== array2.length) {
+    return false;
+  }
+
+  array1.sort();
+  array2.sort();
+
+  for (let i = 0; i < array1.length; i++) {
+    if (array1[i] !== array2[i]) {
+      return false;
+    }
+  }
+
+  return true;
+}
+
+async function deleteEndpoint(rdsClient: RDSClient): Promise<void> {
+  const input = {
+    DBClusterEndpointIdentifier: endpointId
+  };
+  const deleteEndpointCommand = new DeleteDBClusterEndpointCommand(input);
+  try {
+    await rdsClient.send(deleteEndpointCommand);
+  } catch (e: any) {
+    // Custom endpoint already does not exist - do nothing.
+  }
+}
+
+describe("custom endpoint", () => {
+  beforeAll(async () => {
+    env = await TestEnvironment.getCurrent();
+
+    env = await TestEnvironment.getCurrent();
+    const clusterId = env.auroraClusterName;
+    const region = env.region;
+    rdsClient = new RDSClient({ region: region });
+
+    const instances = env.databaseInfo.instances;
+    await createEndpoint(clusterId, instances.slice(0, 1));
+    await waitUntilEndpointAvailable();
+  }, 1000000);
+
+  afterAll(async () => {
+    try {
+      await deleteEndpoint(rdsClient);
+    } finally {
+      rdsClient.destroy();
+    }
+  });
+
+  beforeEach(async () => {
+    auroraTestUtility = new AuroraTestUtility(env.region);
+    driver = DriverHelper.getDriverForDatabaseEngine(env.engine);
+    initClientFunc = DriverHelper.getClient(driver);
+    await ProxyHelper.enableAllConnectivity();
+    await TestEnvironment.verifyClusterStatus();
+    currentWriter = await auroraTestUtility.getClusterWriterInstanceId(env.info.auroraClusterName);
+  });
+
+  afterEach(async () => {
+    if (client !== null) {
+      try {
+        await client.end();
```

Comment:
Do we need to clear the cache each test run via `PluginManager.releaseResources()`

Reply:
Yes, `releaseResources` is called below this line